spec/rubocop/formatter/json_formatter_spec.rb

```diff
@@ -157,5 +157,21 @@
     it 'sets length value for :length key' do
       expect(hash[:length]).to eq(8)
     end
+
+    context 'when the location is pseudo' do
+      let(:location) { RuboCop::Cop::Offense::NO_LOCATION }
+
+      it 'returns a valid hash' do
+        expect(hash).to eq({
+                             start_line: 1,
+                             start_column: 1,
+                             last_line: 1,
+                             last_column: 0,
+                             length: 1,
+                             line: 1,
+                             column: 1
+                           })
```

Comment:
[note] `start_column` is `1`, but `last_column` is `0`. Because this method uses `#real_column` and `#last_column` as follows.
But I feel a bit weird for the difference. Is there no problem?

https://github.com/rubocop-hq/rubocop/blob/751edc7bf3df93d7ec5d59f5ac5b501627a6b723/lib/rubocop/formatter/json_formatter.rb#L68-L76

https://github.com/rubocop-hq/rubocop/blob/751edc7bf3df93d7ec5d59f5ac5b501627a6b723/lib/rubocop/cop/offense.rb#L193-L195

Reply:
This does seem a little odd, however the pseudo source range is also not exactly accurate compared to other offenses since it is used to identify offenses for a file rather than an actual source range. I think it is OK for there to be a bit of non perfect behavior at the moment. It does kind of seem like there might be reason to do a larger refactor to better handle offenses in a file vs offenses in source code.

Maybe the offense itself shouldn't be responsible for the column information, and this kind of info should be handled in the location. I'm not sure how large of a refactor this would take.

Similarly, does it make sense to be setting the line number of the offense to 1 for `NO_LOCATION`? I think it's possible to have a file offense for a file with no code in it. `NO_LOCATION = PseudoSourceRange.new(1, 0, '', 0, 0).freeze`

Reply:
@rrosenblum I agree with your idea, thank you. Surely, in this PR, it seems good that we avoid the big change of the current behavior.

But for the benefit of others, here is an example of comparison:

Current (line=1, col=1):

```
empty_file.rb:1:1: W: Lint/EmptyFile: Empty file detected.
```

Maybe (line=0, col=0):

```
empty_file.rb:0:0: W: Lint/EmptyFile: Empty file detected.
```

I will follow the decision of the community.